search/searcher/search_conjunction_test.go

```diff
@@ -394,6 +394,9 @@ func testScorchCompositeSearchOptimizationsHelper(
 			}
 
 			next, err = cs.Next(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
This one is incorrect (or at least not following the intended pattern).  The err != nil case will exit the for loop.  However, the `err != nil` case should be immediately after the for loop.